cep-XXXX.md

```diff
@@ -0,0 +1,125 @@
+i.e. on the rpath of the executable/shared library that requires the DSO.
+
+The constraints for the DSOs currently in conda-build/rattler-build are,
+
+For Linux
+
+1. If the DSO was in the specified host environment locations ($PREFIX/<rpath>
+   where <rpath>s are listed in `build/rpath`), the DSO has to be in `run` as
+   well; either in the same package or a dependent conda package.
+
+2. If the DSO was in a sysroot ($BUILD_PREFIX/*/sysroot), then these are system
+   DLLs and are allowed. These DSOs usually come from `sysroot_<subdir>`
+   conda packages.
+
+For macOS,
+
+1. If the DSO was in the specified host environment locations ($PREFIX/<rpath>
+   where <rpath>s are listed in `build/rpath`), the DSO has to be in `run` as
+   well; either in the same package or a dependent conda package.
+
+2. If the DSO was in CONDA_BUILD_SYSROOT (the macOS SDK), then these are system
+   DSOs and are allowed.
+
+For windows
+
+1. If the DSO was in the specified host environment locations ($PREFIX/Scripts,
+   $PREFIX/Library/bin, $PREFIX/bin, etc), the DSO has to be in `run` as
+   well; either in the same package or a dependent conda package.
+
+2. If the DSO was in CONDA_BUILD_SYSROOT (this is unset in most scenarios), then   these are system DSOs and are allowed
+
+3. If the DSO was in C:\Windows\System32, then these are system DSOs and are
+   allowed.
+
+4. If the DSO was in DEFAULT_WIN_WHITELIST defined by conda-build/rattler-build,
+   then these are system DSOs and are allowed.
+
+Note that for Linux and macOS, the user has a way of controlling which system
+DSOs are allowed. On Linux, a user can modify the `sysroot_linux-*` package
+and on macOS, the user can specify a custom macOS SDK using `CONDA_BUILD_SYSROOT`.
+However, the user has no way of controlling which system DSO are allowed
+on windows.
+
+### Need for customizability of system DLLs
+
+A user of conda-build/rattler-build might need to customize the set of system DLLs.
+An example from Linux is
+
+  [Removal of libnsl from glibc](https://github.com/conda-forge/linux-sysroot-feedstock/pull/40)
+
+In this case newer versions of glibc dropped libnsl and made it a separate package,
+therefore conda-forge dropped it from `sysroot_linux-*` packages to make sure
+when new conda packages links to libnsl system DLL, conda-build/rattler-build will
+warn/error out.
+
+An example for Windows is,
+
+  [Warn when linking to debug VC runtimes](https://github.com/conda/conda-build/issues/5732)
+
+In this case, on the CI machines we have the debug VC runtime DLLs and are
+considered system DLLs by conda-build/rattler-build, but these debug DLLs
+are not always found in user systems. Also, linking to these debug DLLs
+are not desirable and therefore we do not ship it in our `vc14_runtime`
+conda package. Therefore we need a way to tell conda-build/rattler-build
+on which DLLs are desirable and which are not.
+
+## Specification
```

Comment:
This section should probably mention that these files are expected to be provided by conda packages depended on in `build` or `host`.